development/tools/pkg/jobguard/waiter.go

```diff
@@ -0,0 +1,27 @@
+package jobguard
+
+import (
+	"fmt"
+	"log"
+	"time"
+)
+
+// WaitAtMost executes a function periodically and waits maximum time to finish
+func WaitAtMost(fn func() (bool, error), tickTime time.Duration, duration time.Duration) error {
+	timeout := time.After(duration)
+	tick := time.Tick(tickTime)
+
+	for {
+		ok, err := fn()
```

Comment:
in the current implementation, if a function returns true, we still have to wait for additional tick :(

Reply:
Yup, I used the `waiter` implementation we use everywhere in our components. Of course we can improve it, but I haven't had time for that before my leave.